compiler/src/iree/compiler/Codegen/Common/TransformDialectStrategies.cpp

```diff
@@ -0,0 +1,608 @@
+#include "iree/compiler/Codegen/PassDetail.h"
+#include "iree/compiler/Codegen/Passes.h"
+#include "iree/compiler/Dialect/Flow/IR/FlowDialect.h"
+#include "iree/compiler/Dialect/Flow/IR/FlowOps.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Debug.h"
+#include "mlir/Analysis/SliceAnalysis.h"
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Bufferization/IR/Bufferization.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/GPU/IR/GPUDialect.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/Dialect/Transform/IR/TransformDialect.h"
+#include "mlir/Dialect/Transform/IR/TransformInterfaces.h"
+#include "mlir/Dialect/Transform/IR/TransformOps.h"
+#include "mlir/Dialect/Vector/IR/VectorOps.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/ImplicitLocOpBuilder.h"
+#include "mlir/IR/Location.h"
+#include "mlir/IR/Matchers.h"
+#include "mlir/IR/Types.h"
+#include "mlir/IR/Value.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassRegistry.h"
+namespace mlir {
+namespace iree_compiler {
+
+#define DEBUG_TYPE "iree-transform-builder"
+#define DBGS() (llvm::dbgs() << "[" DEBUG_TYPE "]: ")
+
+using namespace mlir;
+
+// TODO: significantly better namespacing.
+using ::mlir::iree_compiler::IREE::transform_dialect::AllDims;
+using ::mlir::iree_compiler::IREE::transform_dialect::ApplyPatternsOp;
+using ::mlir::iree_compiler::IREE::transform_dialect::ConfigExtractPart;
+using ::mlir::iree_compiler::IREE::transform_dialect::
+    ForeachThreadToWorkgroupOp;
+using ::mlir::iree_compiler::IREE::transform_dialect::IREEBufferizeOp;
+using ::mlir::iree_compiler::IREE::transform_dialect::
+    IREEEraseHALDescriptorTypeFromMemRefOp;
+using ::mlir::iree_compiler::IREE::transform_dialect::IsPermutation;
+using ::mlir::iree_compiler::IREE::transform_dialect::m_StructuredOp;
+using ::mlir::iree_compiler::IREE::transform_dialect::
+    MapNestedForeachThreadToGpuThreadsOp;
+using ::mlir::iree_compiler::IREE::transform_dialect::NumEqualsTo;
+using ::mlir::iree_compiler::IREE::transform_dialect::ShapeKind;
+using ::mlir::iree_compiler::IREE::transform_dialect::StructuredOpMatcher;
+using ::mlir::iree_compiler::IREE::transform_dialect::
+    TileToForeachThreadAndWorkgroupCountRegionOp;
+using ::mlir::iree_compiler::IREE::transform_dialect::
+    VectorToWarpExecuteOnLane0Op;
+using ::mlir::iree_compiler::IREE::transform_dialect::VectorWarpDistributionOp;
+using ::mlir::transform::FuseIntoContainingOp;
+using ::mlir::transform::MatchOp;
+using ::mlir::transform::MergeHandlesOp;
+using ::mlir::transform::PrintOp;
+using ::mlir::transform::SplitHandlesOp;
+using ::mlir::transform::SplitReductionOp;
+using ::mlir::transform::TileToForeachThreadOp;
+using ::mlir::transform::VectorizeOp;
```

Comment:
Why don't we just do `using mlir` and `using mlir::iree_compiler::IREE`?

Reply:
Dropped the ::mlir:: prefix.
I think we still want to avoid hiding the iree_compiler::IREE:: prefix and avoid conflation with the mlir:: prefix